cuda_next/include/detail/dimensions.hpp

```diff
@@ -0,0 +1,109 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Next in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_NEXT_DETAIL_DIMENSIONS
+#define _CUDA_NEXT_DETAIL_DIMENSIONS
+
+#include <cuda/std/mdspan>
+
+namespace cuda_next
+{
+
+template <typename T, size_t... Extents>
+using dimensions = ::cuda::std::extents<T, Extents...>;
+
+// not unsigned because of a bug in ::cuda::std::extents
+using index_type = int;
+
+template <typename T, size_t... Extents>
+struct hierarchy_query_result : public dimensions<T, Extents...>
+{
+  using Dims = dimensions<T, Extents...>;
+  using Dims::Dims;
+  constexpr explicit _CCCL_HOST_DEVICE hierarchy_query_result(const Dims& dims)
```

Comment:
Conventionally the `_CCCL_HOST_DEVICE` comes first, then the explicit
```suggestion
  _CCCL_HOST_DEVICE explicit constexpr hierarchy_query_result(const Dims& dims)
```